Add migration guide for CIS AWS 1.3.0

infraredgirl opened the pull request:
https://gruntwork.atlassian.net/browse/IAC-1641

Remaining work:

- [x] @marinalimeira: [NACL deploy instructions](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386#discussion_r556453607)
- [x] @robmorgan: [`ebs-encryption-multi-region` deploy instructions](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386/files#diff-96ebf8a29723f7ff7440858f0a658156de1fb544a1e8c019e2fb1d32d4112bbcR181)
- [x] @brikis98: Update Acme CIS repos and mention that here in the guide
- [x] Everyone: Double check the compatibility matrix - make sure all the versions are correct
- [x] Everyone: Sanity check the entire document before merge

infraredgirl commented:
Having the version number in the title makes the guide not appear in the list of guides - I guess this is a bug.

bwhaley commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,107 @@
+
+=== Background
+
+The 1.3.0 version of the CIS AWS Foundations Benchmark was release in July of 2020. You can refer to the
```
```suggestion
The 1.3.0 version of the CIS AWS Foundations Benchmark was release in September of 2020. You can refer to the
```
According to [their blog](https://www.cisecurity.org/blog/cis-benchmarks-september-2020-update/), the benchmark was released in September.

bwhaley commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,107 @@
+
+The 1.3.0 version of the CIS AWS Foundations Benchmark was release in July of 2020. You can refer to the
+https://www.cisecurity.org/benchmark/amazon_web_services/[CIS website] where you can download the latest version of the
+Benchmark (as well as all the previous version). The latest version introduces several new reccomendations, and this guide
+will walk you through implementing these new recommendations by utilizing Gruntwork's Infrastructure as Code Library,
```
```suggestion
Benchmark (as well as all the previous version). The latest version introduces several new recommendations, and this guide
will walk you through implementing these with help from the Gruntwork Infrastructure as Code Library,
```

bwhaley commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,107 @@
+
+=== Step 1: Update references to the Gruntwork Infrastructure as Code Library
+
+In order to take advantage of the CIS AWS 1.3.0, you need to update your references to the Gruntwork
```
```suggestion
In order to update to the CIS AWS Foundations Benchmark v1.3.0, you need to update your references to the Gruntwork
```

bwhaley commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,107 @@
+=== Step 1: Update references to the Gruntwork Infrastructure as Code Library
+
+In order to take advantage of the CIS AWS 1.3.0, you need to update your references to the Gruntwork
+Infrastructure as Code Library to use a compatible version. We (Gruntwork) have gone through all our modules in the
+library to test and update the code to be compatible with CIS AWS 1.3.0. As a customer, you need to update to
```
```suggestion
Infrastructure as Code Library to use a compatible version. We (Gruntwork) have reviewed and updated all of the modules in the
library for compatibility with the new version of the Benchmark. As a customer, you need to update to
```

bwhaley reviewed (commented):
Great start! Left a few comments for your consideration on grammar/use of wording. Looking forward to doing a complete review!

brikis98 commented:
> Having the version number in the title makes the guide not appear in the list of guides - I guess this is a bug.

Yes, this is a known issue. I think something in the code struggles with non alphanumeric characters (e.g., decimal points)? @eak12913 @oredavids Any chance of getting this fixed, as it's hard to publish a guide for CIS v1.3.0 without using decimal points 😁

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+
+This guide will walk you through the upgrade process from the 1.2.0 to the 1.3.0 version of the CIS AWS Foundations Benchmark.
+If your infrastructure is already compliant with the 1.2.0 version of the Benchmark, and you are looking to upgrade to 1.3,
+this guide is for you.
```
Perhaps add a sentence like, "If you are starting to work on compliance with this benchmark from scratch, check out our [How to achieve compliance with the CIS AWS Foundations Benchmark](https://gruntwork.io/guides/compliance/how-to-achieve-cis-benchmark-compliance/) guide instead!"

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+
+=== Background
+
+The 1.3.0 version of the CIS AWS Foundations Benchmark was release in September of 2020. You can refer to the
```
```suggestion
The 1.3.0 version of the CIS AWS Foundations Benchmark was released in September of 2020. You can refer to the
```

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+
+The 1.3.0 version of the CIS AWS Foundations Benchmark was release in September of 2020. You can refer to the
+https://www.cisecurity.org/benchmark/amazon_web_services/[CIS website] where you can download the latest version of the
+Benchmark (as well as all the previous version). The latest version introduces several new recommendations, and this guide
```
```suggestion
Benchmark (as well as all the previous versions). The latest version introduces several new recommendations, and this guide
```

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+[[core_concepts]]
+== Core Concepts
+
+=== Background
```
Could you add a section under Core Concepts that summarizes what's new in the 1.3 version of the benchmark? That is, the new requirements added after 1.2? It could be a bullet point list.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+=== Step 1: Update references to the Gruntwork Infrastructure as Code Library
+
+In order to update to the CIS AWS Foundations Benchmark v1.3.0, you need to update your references to the Gruntwork
+Infrastructure as Code Library to use a compatible version. We (Gruntwork) have reviewed and updated all the modules in the
```
```suggestion
Infrastructure as Code Library to use compatible versions. We (Gruntwork) have reviewed and updated all the modules in the
```

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+[[compatibility_table]]
+[cols="1,1h,1"]
+|===
+|Gruntwork Repo |Minimum version with CIS AWS 1.3.0 support |Corresponding CIS AWS 1.3.0 recommendations
```
It took me a minute to understand what the "Corresponding CIS AWS 1.3.0 recommendations" were. Adding these new items to the Core Concepts section as mentioned above would definitely help.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+The following table provides a summary of all the relevant Gruntwork AWS modules and the respective versions that are
+compatible with CIS AWS 1.3.0:
+
+[[compatibility_table]]
```
Reminder to update these versions when we complete the last few items for 1.3 (e.g., the NACLs stuff).

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,124 @@
+|===
+
+
+=== Step 2: Manual steps
```
I think there's quite a bit more to do than this... Many of the requirements require deploying new modules that customers would not have been running at all previously, so bumping version numbers alone is not enough. Going off the list in [this Jira epic](https://gruntwork.atlassian.net/browse/IAC-1604):

1. You need to deploy the IAM access analyzer module.
2. You have to request increased NACL limits before deploying the new NACL stuff.
3. You have to deploy the new `cleanup-expired-certs` module.
4. You have to deploy the new `ebs-encryption-multi-region` module.

There may be others, so make sure all the people that worked on these parts take a careful look!

infraredgirl replied:
Yeah, definitely, that's why this PR is still WIP. Everyone who worked on the CIS upgrade is supposed to add their bits to this guide.

ina-stoyanova replied:
I'm wondering 🤔  besides the tickets we had as part of this epic in Jira (https://gruntwork.atlassian.net/browse/IAC-1604), do we have another way of gathering data when people have done work towards CIS 1.3 outside of the project itself (e.g. the `ebs-encryption-multi-region`)? I tried searching slack, but couldn't really dig out anything else really

brikis98 replied:
Hm, not really...

@bwhaley When you went through the new requirements for v1.3, did you compare it to just v1.2, and create tickets for each new item in v1.3? Or did you compare it to _what we already had at Gruntwork_, and only create tickets for the new items we'd have to create at Gruntwork? If it's the latter, then there are new modules or features we added since we released our 1.2 offering that we need to explicitly tell customers about in this upgrade guide?

ina-stoyanova replied:
I was curious to have a play around with this bug, and I have to say it's very annoying and unfortunately my lack of experience with Jekyll and playing around didn't really help!

Anyway, I thought I'd share a few things I looked into, and hopefully it could be a very simple fix when combined with more knowledgeable people in how this works! 

https://jekyllrb.com/docs/liquid/filters/
https://shopify.github.io/liquid/filters/escape/

Basically, as far as I could gather, none of the `;`, `,`, `.`, `:` characters work, so I assume it's some sort of escaping of them that needs to happen!

bwhaley replied:
I looked at the Benchmark v1.3.0, compared each control to 1.2, and opened a ticket for each difference. I do not think there are any modules or steps that we need to cover that are not in this epic. For instance, the EBS encryption module was already underway when I put the epic together, and it was released before this project began, but it's still mentioned here in a ticket. Hope that makes sense - let me know if it doesn't!

ina-stoyanova replied:
Ahaa! That makes sense, @bwhaley :) Thanks for confirming - I can see that the EBS is indeed under the Jira epic, but last time I must have missed it or something, so my bad here! 

Ok, so in this case we're all good and confident all that's on Jira is indeed the correct list of changes we need to mention 🙂

brikis98 reviewed (commented):
Thanks for putting this together!

Having seen this, I think the right thing to do is to update the Acme CIS repos too, so customers have a concrete upgrade example to look at. We need to add that to the Jira epic.

ina-stoyanova commented:
Thanks for making a start on this @infraredgirl 🙂 I"m having a look to add bits about IAM access analyzer now

eak12913 commented:
> > Having the version number in the title makes the guide not appear in the list of guides - I guess this is a bug.
> 
> Yes, this is a known issue. I think something in the code struggles with non alphanumeric characters (e.g., decimal points)? @eak12913 @oredavids Any chance of getting this fixed, as it's hard to publish a guide for CIS v1.3.0 without using decimal points 😁

Would this be a blocker for you folks? If at all possible - I'd prefer to not switch context right now but I am worried that this will hold y'all up.

The last time I looked at the code it was somewhat of a tangled mess of wires with at some amount of assumptions baked in about what Jekyll will do too (in addition to whatever our mess of logic is doing). In order to dive into that - I'd have to page out what I'm working on now.

brikis98 commented:
> > > Having the version number in the title makes the guide not appear in the list of guides - I guess this is a bug.
> > 
> > 
> > Yes, this is a known issue. I think something in the code struggles with non alphanumeric characters (e.g., decimal points)? @eak12913 @oredavids Any chance of getting this fixed, as it's hard to publish a guide for CIS v1.3.0 without using decimal points 😁
> 
> Would this be a blocker for you folks? If at all possible - I'd prefer to not switch context right now but I am worried that this will hold y'all up.
> 
> The last time I looked at the code it was somewhat of a tangled mess of wires with at some amount of assumptions baked in about what Jekyll will do too (in addition to whatever our mess of logic is doing). In order to dive into that - I'd have to page out what I'm working on now.

I think the guide needs to have "1.2" and "1.3" in the title. If there's some workaround way to get that without forcing you to context switch, we're happy to use that... But otherwise, this won't have the right title, or the right SEO benefits, and changing it later would change the URL, which would require redirects, which we'd prob forget, leading to 404s... So, yea, I think we need _some_ way to have the proper title on this guide.

ina-stoyanova commented:
> > > > Having the version number in the title makes the guide not appear in the list of guides - I guess this is a bug.
> > > 
> > > 
> > > Yes, this is a known issue. I think something in the code struggles with non alphanumeric characters (e.g., decimal points)? @eak12913 @oredavids Any chance of getting this fixed, as it's hard to publish a guide for CIS v1.3.0 without using decimal points 😁
> > 
> > 
> > Would this be a blocker for you folks? If at all possible - I'd prefer to not switch context right now but I am worried that this will hold y'all up.
> > The last time I looked at the code it was somewhat of a tangled mess of wires with at some amount of assumptions baked in about what Jekyll will do too (in addition to whatever our mess of logic is doing). In order to dive into that - I'd have to page out what I'm working on now.
> 
> I think the guide needs to have "1.2" and "1.3" in the title. If there's some workaround way to get that without forcing you to context switch, we're happy to use that... But otherwise, this won't have the right title, or the right SEO benefits, and changing it later would change the URL, which would require redirects, which we'd prob forget, leading to 404s... So, yea, I think we need _some_ way to have the proper title on this guide.

I was curious to have a play around with this bug, and I have to say it's very annoying and unfortunately my lack of experience with Jekyll and playing around didn't really help!

Anyway, I thought I'd share a few things I looked into, and hopefully it could be a very simple fix when combined with more knowledgeable people in how this works! 

https://jekyllrb.com/docs/liquid/filters/
https://shopify.github.io/liquid/filters/escape/

Basically, as far as I could gather, none of the `;`, `,`, `.`, `:` characters work, so I assume it's some sort of escaping of them that needs to happen!

eak12913 commented:
Thanks for taking a look @ina-stoyanova! - I will just finish up a few things and can try to take a look as well.  If memory serves me correctly from last time, the problem is somewhere in the interplay between the following things:
- The title of the article
- The search behavior which is responsible for showing/hiding the main sections on that index page
- Scroll-spy - which is a library that's linking up the nav on the left with what's displayed in the main section

There's interplay between these things and the last time I looked - I _was_ able to figure out Jim's issue - but it still didn't solve this one.

If you'd like - we can pair code? I'm no Jekyll expert and only have cursory exposure to this code from the troubleshooting I did on it last time. Otherwise I'll take it up either later today (probably too late for you) or Monday depending on how today's other tasks go.

ina-stoyanova commented:
> Thanks for taking a look @ina-stoyanova! - I will just finish up a few things and can try to take a look as well. If memory serves me correctly from last time, the problem is somewhere in the interplay between the following things:
> 
> * The title of the article
> * The search behavior which is responsible for showing/hiding the main sections on that index page
> * Scroll-spy - which is a library that's linking up the nav on the left with what's displayed in the main section
> 
> There's interplay between these things and the last time I looked - I _was_ able to figure out Jim's issue - but it still didn't solve this one.
> 
> If you'd like - we can pair code? I'm no Jekyll expert and only have cursory exposure to this code from the troubleshooting I did on it last time. Otherwise I'll take it up either later today (probably too late for you) or Monday depending on how today's other tasks go.

Yeah - I'd love to pair code if it's on Monday. Although, it's not my domain - it would certainly be useful to pair code with you! Either way - thanks for filling in my gaps :) I knew there must be more to it, as simple fixes obviously didn't work 😜Loads to learn as always! Feel free to pick it up whenever's convenient for you - but if Monday, I'll make time to work with you :)

ina-stoyanova commented:
Just trying to understand what's still left on this PR? Is the following correct?

[ ] Fix title bug (not showing '1.3' in the guide title)
[ ] Add section for EBS encryption
[ ] Add section for NACL
[ ] Add section for object-level logging functionality
[ ] Add section for log metrics & alarms 
[ ] Add section S3 bucket (using private buckets)
[ ] Review section for IAM access analyzer (some bits added, but definitely still needs review)

N.B.: This PR could be reviewed, but not merged until this is also done: https://gruntwork.atlassian.net/browse/IAC-1683

<img width="698" alt="Screenshot 2021-01-11 at 12 27 08" src="https://user-images.githubusercontent.com/32835571/104169598-7c7d7380-5408-11eb-98a3-730f2169fa37.png">

infraredgirl commented:
Do we need to have dedicated section for the items that were just upgrades to the existing modules? I understand the need for dedicated sections for new modules, but I'm not sure that the S3 bucket upgrades (which are just refactors to use a different modules under the hood) need a section each?

eak12913 commented:
@infraredgirl Could you please try to change your title to what you originally wanted it to be and double check. @ina-stoyanova and I have fixed the issue and we believe that it should no longer be a problem.

infraredgirl commented:
> @infraredgirl Could you please try to change your title to what you originally wanted it to be and double check. @ina-stoyanova and I have fixed the issue and we believe that it should no longer be a problem.

Works now! Thanks so much for the fix!

eak12913 commented:
I was originally dreading this one because I thought it was going to be a can of worms. It ended up being pretty straight forward and to correct my earlier assumption, had nothing to do with ScrollSpy or the search logic. The issue was with the `id` property. We were basing the `id` property off of the `title` of the article but were not `slugifying`.

ina-stoyanova commented:
> Do we need to have dedicated section for the items that were just upgrades to the existing modules? I understand the need for dedicated sections for new modules, but I'm not sure that e.g. the S3 bucket upgrades (which are just refactors to use a different modules under the hood) need a section each?

Ahaa! Yeah - I wasn't sure at all what was the intention. You've got a very good point that if it's just an upgrade - maybe doesn't deserve a section at all. What were you thinking. I'm following your lead here, just wanted to see where we are roughly with it 🙂 Hope that's ok!

infraredgirl commented:
> Ahaa! Yeah - I wasn't sure at all what was the intention. You've got a very good point that if it's just an upgrade - maybe doesn't deserve a section at all. What were you thinking. I'm following your lead here, just wanted to see where we are roughly with it  Hope that's ok!
It might make sense to have a section for each newly added module and another single section to mention all updates to the existing modules. WDYT?

ina-stoyanova commented:
> It might make sense to have a section for each newly added module and another single section to mention all updates to the existing modules. WDYT?

yep - good with that!

infraredgirl commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+To help you achieve CIS 1.3 compliance, you can find examples of how to use the ready-made module as part of the `landingzone` module in the `aws-service-catalog` repository https://github.com/gruntwork-io/aws-service-catalog/releases/tag/v0.15.4[v0.15.4], and specifically the `account-baseline-root` and `account-baseline-security` https://github.com/gruntwork-io/aws-service-catalog/tree/v0.15.4/ ...
+
+==== 2. Deploy ... module (recommendation 1.19)
+#TODO(Rob) IAC-1606
```
@robmorgan please add deploy instructions here.

infraredgirl commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+#TODO(Rob) IAC-1606
+
+==== 3. Deploy ... module (recommendation 5.1)
+#TODO(Marina) IAC-1613
```
@marinalimeira please add deploy instructions here.

marinalimeira replied:
Done!

infraredgirl commented:
I added a couple of paragraphs that talk about new vs existing modules - feedback welcome if this makes sense or there's a better way to cover this.

I also added sections for the two remaining new modules we need to cover (let me know if I missed any new modules!). Rob and Marina should add their deployment instructions there.

Hope all that makes sense, as always I'm open to feedback if anyone can think of a better way to break down this information!

robmorgan commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+The 1.3.0 version of the CIS AWS Foundations Benchmark was released in September of 2020. You can refer to the
+https://www.cisecurity.org/benchmark/amazon_web_services/[CIS website] where you can download the latest version of the
+Benchmark (as well as all the previous versions). The latest version introduces several new recommendations, and this guide
+will walk you through implementing these with help from the Gruntwork's Infrastructure as Code Library,
```
```suggestion
will walk you through implementing these using Gruntwork's Infrastructure as Code Library,
```
Does this read better? I guess our IaC library is technically helping, but I feel most users will directly consume the code modules themselves vs using them as a guide.

robmorgan commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+https://www.cisecurity.org/benchmark/amazon_web_services/[CIS website] where you can download the latest version of the
+Benchmark (as well as all the previous versions). The latest version introduces several new recommendations, and this guide
+will walk you through implementing these with help from the Gruntwork's Infrastructure as Code Library,
+so that your code becomes fully compliant with the 1.3.0 version of the framework.
```
```suggestion
so that your infrastructure is fully compliant with version 1.3.0 of the framework.
```
Also, we are technically targeting CIS compliance for "infra" versuses "code" right? I think our solution can be used to buy compliant infrastructure, hence code, but the end customer must look after the physical controls themselves.

robmorgan commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+so that your code becomes fully compliant with the 1.3.0 version of the framework.
+
+==== New recommendations
+These are the new recommendations introduced in the version 1.3.0 of the Benchmark. You can think of these as a "diff"
```
```suggestion
These are the new recommendations introduced in version 1.3.0 of the Benchmark. You can think of these as a "diff"
```
Grammar suggestion

robmorgan commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,178 @@
+==== New Gruntwork modules vs existing modules
+To achieve compliance with the newly added 10 recommendations listed above, we created three new modules in the
+Gruntwork's Infrastructure as Code Library, and updated a bunch of existing modules. Namely: recommendations 1.19, 1.21
+and 5.1 required creation of new modules; the rest of recommendations were achieved by updating existing modules.
```
```suggestion
and 5.1 required the creation of new modules; the rest of the recommendations were achieved by updating existing modules.
```

robmorgan commented:
Minor note: in my contributions, I'm adding links to the [cis-compliance-aws](https://github.com/gruntwork-io/cis-compliance-aws) repo, which I believe we are going to rename as part of our next batch of work. We'll need to come back later and update the links in this document although I don't believe they'll actually break with the GitHub auto-redirect feature.

infraredgirl commented:
Apologies - I had overlooked that the new EBS encryption module also needs deployment instructions. I added a section for that in the latest commit. @robmorgan would you mind filling that in?

I also updated the PR description to track what work we have remaining - please LMK if anything is missing there!

robmorgan commented:
@infraredgirl sure! I've filled in the EBS volume encryption section. I've also reworded some of the paragraphs to read better as I noticed you re-ordered them 👍

brikis98 commented:
> * [ ]  @brikis98: Update Acme CIS repos and mention that here in the guide

Starting work on this now.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,213 @@
+|https://github.com/gruntwork-io/package-openvpn/releases/tag/v0.13.0[v0.13.0]
+|1.20, 2.1.1, 2.1.2
+
+|cis-compliance-aws
+|https://github.com/gruntwork-io/cis-compliance-aws/releases/tag/v0.9.2[v0.9.2]
```
I think this should be 0.9.3? https://github.com/gruntwork-io/cis-compliance-aws/releases/tag/v0.9.3

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
````diff
@@ -0,0 +1,213 @@
+as `module-vpc`). There is one new required argument, `allow_administrative_remote_access_cidrs`. You will add the
+CIDRs from your office!
+
+```
````
```suggestion
```hcl
```

infraredgirl replied:
Fixed this to use the correct adoc code block syntax.

brikis98 reviewed (commented):
Went through the guide, and overall, it looks really solid 👍 

I commented on a couple minor issues. See also https://github.com/gruntwork-io/usage-patterns/pull/514.

robmorgan commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+EC2 supports encryption at rest when using the Elastic Block Store (EBS) service. While disabled by default, forcing encryption when creating EBS volumes is supported. Encrypting data at rest reduces the likelihood that it is
+unintentionally exposed and can nullify the impact of disclosure if the encryption remains unbroken. Recommendation 2.2.1 specifies a manual process to encrypt EBS volumes using the AWS Console, however Gruntwork has developed
+a module that configures volume encryption by default in all enabled regions. Check out the https://github.com/gruntwork-io/module-security/tree/master/modules/ebs-encryption-multi-region[ebs-encryption-multi-region] to
+configure AWS EBS encryption in all enabled regions for an AWS Account.
```
```suggestion
configure AWS EBS encryption in all enabled regions of an AWS Account.
```

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+|https://github.com/gruntwork-io/package-openvpn/releases/tag/v0.13.0[v0.13.0]
+|1.20, 2.1.1, 2.1.2
+
+|cis-compliance-aws
+|https://github.com/gruntwork-io/cis-compliance-aws/releases/tag/v0.9.3[v0.9.3]
```
This now needs to be [v0.10.0](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/releases/tag/v0.10.0).

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+|===
+|Gruntwork Repo |Minimum version with CIS AWS 1.3.0 support |Corresponding CIS AWS 1.3.0 recommendations
+
+|module-security
```
NIT: we renamed all these repos to the `terraform-aws-xxx` format.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+
+  # ... the existing variables weren't affected!
+
+  allow_administrative_remote_access_cidrs = {
+    berlin_office = "1.2.3.4/32"
+    ny_office     = "6.7.8.9/32"
+  }
```
This example needs to be updated to use 3 input variables, one per subnet tier:

- `allow_administrative_remote_access_cidrs_public_subnets`
- `allow_administrative_remote_access_cidrs_private_app_subnets`
- `allow_administrative_remote_access_cidrs_private_persistence_subnets`

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+[source,hcl]
+----
+module "vpc_app_network_acls" {
+  source = "git::git@github.com:gruntwork-io/cis-compliance-aws.git//modules/vpc-app-network-acls?ref=v0.9.3"
```
```suggestion
  source = "git::git@github.com:gruntwork-io/cis-compliance-aws.git//modules/vpc-app-network-acls?ref=v0.10.0"
```

brikis98 commented:
Updated for Acme CIS repos are here:

https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/6
https://github.com/gruntwork-io/cis-infrastructure-live-acme/pull/8

infraredgirl commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+|1.20, 2.1.1, 2.1.2
+
+|terraform-aws-vpc
+|https://github.com/gruntwork-io/terraform-aws-vpc/releases/tag/v0.12.4[v0.12.4]
```
I bumped this to 0.12.4 as it looks like the minor update are relevant to the new vpc modules. LMK if this is not right.

ina-stoyanova replied:
Updating this to 0.13.0 now? The latest release seems to fix a bug with the NACLs module. @marinalimeira is this correct?

infraredgirl replied:
Good catch! I committed your suggestion.

infraredgirl commented:
> Updated for Acme CIS repos are here:
> 
> [gruntwork-io/cis-infrastructure-modules-acme#6](https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/6)
> [gruntwork-io/cis-infrastructure-live-acme#8](https://github.com/gruntwork-io/cis-infrastructure-live-acme/pull/8)

Should we make a mention of these in the guide?

infraredgirl commented:
> > > Updated for Acme CIS repos are here:
> > > [gruntwork-io/cis-infrastructure-modules-acme#6](https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/6)
> > > [gruntwork-io/cis-infrastructure-live-acme#8](https://github.com/gruntwork-io/cis-infrastructure-live-acme/pull/8)
> > 
> > 
> > Should we make a mention of these in the guide?
> 
> Yes! We should make these a first-class part of the guide: "See the Acme CIS examples to see what the code changes may look like."

Added in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386/commits/e80a4ee74d6a4d47a44a44fc549cffcc8e65ca65. I've linked directly to the update PRs - if there's a better location to link to please LMK!

infraredgirl commented:
I think we can open this up for reviews!

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+[[deployment_walkthrough]]
+== Deployment walkthrough
+
+Please follow the steps below to upgrade from the version 1.2.0 to 1.3.0 of the Benchmark. To see the actual relevant code
+changes, please refer to the Acme CIS examples https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/6[here]
+and https://github.com/gruntwork-io/cis-infrastructure-live-acme/pull/8[here].
```
I actually missed these references to Acme initially. Since they are _very_ useful for doing this upgrade, perhaps put them in a call-out, and use bullet points for the two PR links?

infraredgirl replied:
By call-out, do you mean a `NOTE` block? If so, I've updated it in https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386/commits/801a74358dd3232cf824e34e20ac221860b3d364. LMK if you meant something else!

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+
+==== 4. Deploy the vpc-app-network-acl and vpc-mgmt-network-acl modules (recommendation 5.1)
+
+If you are already using the `vpc-app-network-acl` and/or `vpc-mgmt-network-acl` modules, you need to update the
```
NIT: add a sentence of context here about how we added a new `vpc-app-network-acl` module to the CIS repo, how this module enforces rules around SSH/RDP access, etc, and how customers will need to deploy this new module. _Then_ you can go into the part about what to do if you are using the `vpc-app-network-acl` module from `terraform-aws-vpc` already.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+
+If you are already using the `vpc-app-network-acl` and/or `vpc-mgmt-network-acl` modules, you need to update the
+source URL from the https://github.com/gruntwork-io/terraform-aws-vpc[`terraform-aws-vpc` module] (previously known
+as `module-vpc`). There is one new required argument, `allow_administrative_remote_access_cidrs`. You will add the
```
This is now out of date! There are 3 new arguments:

* `allow_administrative_remote_access_cidrs_public_subnets`
* `allow_administrative_remote_access_cidrs_private_app_subnets`
* `allow_administrative_remote_access_cidrs_private_persistence_subnets`

brikis98 replied:
Oh, and it should explain the difference between the three: e.g., the `public_subnets` one is for servers in the public subnets, and usually set to the CIDRs of your office; the other ones are for private subnets and usually set to the CIDRs of the VPC or specific subnets within that VPC.

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+  allow_administrative_remote_access_cidrs_private_app_subnets         = { app_vpc_cidrs = module.app_vpc.vpc_cidr_block }
+  allow_administrative_remote_access_cidrs_private_persistence_subnets = { app_vpc_cidrs = module.app_vpc.vpc_cidr_block }
+}
+----
```
If you're switching from `vpc-app-network-acls` in `module-vpc` to this new one, you'll also need to run `state mv` commands! As all the original resources are now nested one module deeper.

marinalimeira replied:
script here: https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/7

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+}
+----
+
+Note that these new Network ACL Rules exceed the default AWS Quota for NACL Rules. We created a Terraform module
+(https://github.com/gruntwork-io/terraform-aws-utilities/tree/master/modules/request-quota-increase[`request-quota-increase`])
+where you can request a quota increase. If you use two remote administration ports (the defaults for both modules
+are 22 (SSH) and 3389 (Remote Desktop)), you can add up to 10 CIDRs. Check out
+https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/v0.10.0/modules/vpc-app-network-acls#calculating-nacl-rules-limits[our
+docs] to see how to calculate the maximum number of CIDRs that you can add.
```
This should be a separate step to make it clearer.

brikis98 reviewed (commented):
Getting super close!

brikis98 commented:
Yup, that's what I meant!

On Tue, Jan 26, 2021 at 3:04 PM Ana Krivokapić <notifications@github.com>
wrote:

> *@infraredgirl* commented on this pull request.
> ------------------------------
>
> In _posts/2021-01-06-how-to-update-to-cis-13.adoc
> <https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386#discussion_r564580801>
> :
>
> > +Please follow the steps below to upgrade from the version 1.2.0 to 1.3.0 of the Benchmark. To see the actual relevant code
> +changes, please refer to the Acme CIS examples https://github.com/gruntwork-io/cis-infrastructure-modules-acme/pull/6[here]
> +and https://github.com/gruntwork-io/cis-infrastructure-live-acme/pull/8[here].
>
> By call-out, do you mean a NOTE block? If so, I've updated it in 801a743
> <https://github.com/gruntwork-io/gruntwork-io.github.io/commit/801a74358dd3232cf824e34e20ac221860b3d364>.
> LMK if you meant something else!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gruntwork-io/gruntwork-io.github.io/pull/386#discussion_r564580801>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFNZZH37L5X3R4RQJYOZLLS33KW5ANCNFSM4VXYZMQA>
> .
>

ina-stoyanova commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+|1.20, 2.1.1, 2.1.2
+
+|terraform-aws-vpc
+|https://github.com/gruntwork-io/terraform-aws-vpc/releases/tag/v0.12.4[v0.12.4]
```
```suggestion
|https://github.com/gruntwork-io/terraform-aws-vpc/releases/tag/v0.13.0[v0.13.0]
```

ina-stoyanova commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,214 @@
+configure AWS EBS encryption in all enabled regions of an AWS Account.
+
+==== 4. Deploy the vpc-app-network-acl and vpc-mgmt-network-acl modules (recommendation 5.1)
+
```
```suggestion
To help us achieve CIS 1.3 compliance, we've also created the `vpc-app-network-acl` module in our [dedicated CIS service catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/modules/vpc-app-network-acls). This module is designed in a way that follows CIS 1.3 recommendations - restrict access by default, but only allow explicitly listed SSH and RDP connections and hosts. To be compliant, you'll need to deploy the new module. For more details, please refer to the [dedicated module README](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/modules/vpc-app-network-acls/README.md) and [the relevant example](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/examples/vpc-network-acls). 
```

ina-stoyanova commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+|Gruntwork Repo |Minimum version with CIS AWS 1.3.0 support |Corresponding CIS AWS 1.3.0 recommendations
+
+|terraform-aws-security
+|https://github.com/gruntwork-io/terraform-aws-security/releases/tag/v0.44.6[v0.44.6]
```
```suggestion
|https://github.com/gruntwork-io/terraform-aws-security/releases/tag/v0.44.10[v0.44.10]
```

infraredgirl replied:
I don't think these updates are relevant to CIS?

ina-stoyanova replied:
You're right!! I failed to see it was the rename. And also your comment below for keeping to the minimal version that is relevant is applicable here too!

ina-stoyanova commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+|1.20, 1.21, 2.1.1, 2.1.2, 3.10, 3.11
+
+|terraform-aws-monitoring
+|https://github.com/gruntwork-io/terraform-aws-monitoring/releases/tag/v0.24.0[v0.24.0]
```
```suggestion
|https://github.com/gruntwork-io/terraform-aws-monitoring/releases/tag/v0.24.1[v0.24.1]
```

infraredgirl replied:
I don't think these updates are relevant to CIS?

ina-stoyanova replied:
You're right!! I failed to see it was the rename. And also your comment below for keeping to the minimal version that is relevant is applicable here too!

ina-stoyanova commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,220 @@
+|3.10, 3.11, 4.15, 5.1
+
+|terraform-aws-service-catalog
+|https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.15.4[v0.15.4]
```
not really sure what to do here - we've got this version released already, which is backwards incompatible. But, it also is not directly related to CIS 1.3 - should we link to the latest working release, or the one that's relevant and got the changes for CIS?

infraredgirl replied:
I think we should go with the minimum release that has the relevant CIS changes.

marinalimeira replied:
I don't think we launched this to the customers yet, so I don't think it is an issue.

ina-stoyanova commented:
Hey @marinalimeira @infraredgirl I believe with the last few comments, this is ready to go?

brikis98 commented on _posts/2021-01-06-how-to-update-to-cis-13.adoc:
```diff
@@ -0,0 +1,232 @@
+
+The new CIS AWS v1.3 recommendations require that all expired SSL/TLS certificates stored in AWS IAM are automatically removed. Removing expired SSL/TLS certificates eliminates the risk that an invalid certificate will be deployed
+accidentally to a resource such as AWS Elastic Load Balancer (ELB), which can damage the credibility of the application/website behind the ELB. As a best practice, it is recommended to delete expired certificates. To help you
+achieve this recommendation, check out the https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/examples/cleanup-expired-certs/terraform[example] of the ready-made `cleanup-expired-certs` module.
```
You can also cross-link to the 1.3.0 guide, as it now has a section on how to use this module (with Terragrunt), and similar sections for each module below!

infraredgirl replied:
Added in 8ada2d539f0d5988ff84905101214d750accfb40. (Note that these links don't resolve right now because the 1.3.0 guide hasn't been merged yet!)

infraredgirl replied:
Added in 8ada2d539f0d5988ff84905101214d750accfb40. (Note that these links don't resolve right now because the 1.3.0 guide hasn't been merged yet!)

brikis98 reviewed (approved):
This is also good to go after a few minor wording / typo tweaks!

infraredgirl commented:
Woohoo! 

Many thanks to everyone who contributed to this PR, both in terms of content and reviews!

Merging now!
